oeps/oep-0001.rst

```diff
@@ -0,0 +1,335 @@
++--------+---------------------------------------+
+|Created | 26-Mar-2016                           |
++--------+---------------------------------------+
+|Arbiter | Eddie Fagin <efagin@edx.org>          |
++--------+---------------------------------------+
+
+
+What is an OEP?
+===============
+
+OEP (pronounced "oh-epp") stands for Open edX (Enhancement) Proposal. An OEP is a
+document that specifies a specific choice being made by the
+Open edX community, in the form of a best-practice, architecture
+design, technology decision, or process adjustment. It should
+provide the use-cases and rationales that surround that choice.
+OEPs are not the only way for a change to be made to Open edX,
+however the goal is for it to be a repository collecting the
+history of large and/or broadly relevant choices. 
+
+OEP Types
+=========
+* A Process proposal describes a change to how the Open edX community
+  functions.
+* A "Best Practices" proposal describes a technology or implementation
+  choice that the Open edX community believes all Open edX services
+  and/or libraries should use or follow.
+* An Architecture proposal describes relationships between Open edX
+  services and libraries, including their APIs and data formats.
+
+Workflow
+========
+
+Submitting an OEP
+-----------------
+The OEP process begins with a new idea for Open edX. It is highly
+recommended that a single OEP contain a single key proposal or new idea.
+Small enhancements or patches often don't need an OEP and can be injected
+into the Open edX development workflow with a patch submission.
+
+Each OEP must have a Author -- someone who writes the OEP using the
+style and format described below, shepherds the discussions in the
+appropriate forums, and attempts to build community consensus around
+the idea. The OEP Author should first attempt to
+ascertain whether the idea is appropriate for an OEP. Posting to the
+`edx-code`_ mailing list, ``arch@edx.org`` and ``archexternal@edx.org``
+is the best way to go about this.
+
+Vetting an idea publicly before going as far as writing an OEP is meant
+to save the potential author time. Many ideas have been brought forward
+for changing Open edX that have been rejected for various reasons. Asking
+the Open edX community first if an idea is original helps prevent too much
+time being spent on something that is guaranteed to be rejected based on
+prior discussions (searching the internet does not always do the trick).
+It also helps to make sure the idea is applicable to the entire community
+and not just the author. Just because an idea sounds good to the author
+does not mean it will work for most Open edX installations.
+
+Once the Author has asked the Open edX community as to whether an idea
+has any chance of acceptance, a draft OEP should be submitted as a pull request
+against the `central OEP repository`_. The pull request title must start
+with "OEP-XXX", and claims that OEP number for the included proposal.
```

Comment:
where XXX is a number that is automatically generated? manually selected? by the author...